### ISSUE TITLE
Issue with layouts renderer and sections => merge sections in one view

### ISSUE THREAD

**Issue by RedskyThirty** [codeigniter4/CodeIgniter4#2491]
Hello,

It seems that there is a problem with the layouts and I think that this is due to "ob_start / ob_flush / ..." in the renderer.
It will be easier to explain my problem with an example...

```
public function send(): string
{
    $email = view('email');
   
    // Send email script here...

    return view('confirmation');
} 
```

View "email" extends a layout "x" with a section called "body".
View "confirmation" extends another layout "y" with a section called "body".

Sometimes (but not always), "view('confirmation')" displays the "body" section of "email" and "confirmation". The renderer merges both sections even if the layout is not the same.
This is really annoying. For the moment, I try to prefix all my sections in each layouts but it would be better if I didn't have to do this kind of things.

**Comment by najdanovicivan** [comment]
Can you provide view files you're using.  View are rendered using the sharedRenderer service. I'm noticing some issues with properties $sections and $layout not being reset after outputting the view.

I'm mostly interested which name you're using for section.

**Comment by RedskyThirty** [comment]
I'm sorry, I cannot share my real files but here is what they look like:

**Layout 01**
```
Some HTML here...
<?= $this->renderSection('body') ?>
Some HTML here...
```

**View 01**
```
<?= $this->extend('layouts/layout_01') ?>
<?= $this->section('body') ?>
Some HTML here...
<?= $this->endSection() ?>
```

**Layout 02**
```
Some HTML here (different from layout 01)...
<?= $this->renderSection('body') ?>
Some HTML here (different from layout 01)...
```

**View 02**
```
<?= $this->extend('layouts/layout_02') ?>
<?= $this->section('body') ?>
Some HTML here...
<?= $this->endSection() ?>
```

**Comment by RedskyThirty** [comment]
FYI, I renamed "body" by something like "l1_body" and "l2_body" to prevent the issue.

**Comment by najdanovicivan** [comment]
Yes. Exactly what I'was taking about. As sections are not being reset on the end of rendering having  sections with the same name in multiple layouts can cause issues.

**Comment by RedskyThirty** [comment]
Yes, but am I wrong if I say that for the moment the framework has no method to finalize a "renderSection()"?

**Comment by najdanovicivan** [comment]
I've submitted PR for this one. I've only added one line in the code. Can you manually paste 

```
// Reset sections
$this->sections = [];

```
 
After  line 249 in system/View/View.php to see if it fixes the issue for you ?

**Comment by MGatner** [comment]
We’ll have to wait on the test outcome but this looks like a good and isolated change so I’m glad to merge shortly.

**Comment by MGatner** [comment]
Oops, GitHub auto-closed. @RedskyThirty can you confirm if this is resolved?

**Comment by RedskyThirty** [comment]
I don't have time to test now but I suppose it's ok ;-)

**Comment by najdanovicivan** [comment]
@MGatner auto close happens because I put  Fixes prefix in PR message

```
Fixes #2491
```

So once PR is merged the Issue gets closed automatically. I'm gonna send future PRs with Reference only

**Comment by MGatner** [comment]
This was causing other issues that we don't have tests in place for; I had to revert the change for now until we can look into it.

**Comment by najdanovicivan** [comment]
@MGatner can you provide me a sample where the issue happens so that I can create test and then check the issue in more detail?

**Comment by MGatner** [comment]
Sure. It happened on this project: https://github.com/MGatner/nexus

Notice that the layout has its own sub-calls to views, which I think is where the issue is. When the parser clears the sections those nested calls don’t resolve.

**Comment by RedskyThirty** [comment]
Here you can find a simple demo of the issue ;-)
[https://www.dropbox.com/sh/mxdpz6fd35txrpw/AACWbBPE4VlUENOjueImTIhJa?dl=0](https://www.dropbox.com/sh/mxdpz6fd35txrpw/AACWbBPE4VlUENOjueImTIhJa)

**Comment by najdanovicivan** [comment]
@MGatner  and @RedskyThirty  I've submitted new PR which takes a different approach on this one. Additionally I've included the unit test for the mentioned issue. it will be great if you can do some testing on your end.
